### PR TITLE
Inputs for sliders

### PR DESCRIPTION
**Pull request by will-moore**
See https://trello.com/c/AozomHkL/125-sliders-allow-typing-in-values

For the channel sliders, we now allow users to type in the value they want (as in webclient & Insight).
This allows greater accuracy and you can also enter values outside of the slider range.
We now show the existing Average value (in light grey) if selected images have different values for start/end.
This allows user to edit existing value instead of '-' as before.
We use "number" fields, so you can't ender non-numeric characters. Most browsers also support mousewheel scrolling to edit the value.

To test:
  Select a single image or multiple images:
- Where start/end values for a channel are same in all images, text is black.
- Editing values with slider of by typing in input and hitting enter will update value in all images to the same value (text will become black).
- Can input values outside of slider range. Slider range should expand as needed.
- Check that you can't input a "start" value greater than the "end" value and vice versa (will get alert message on enter).

screen shot 2016-08-22 at 11 50 42

### CODE REVIEW

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [FIGURE-merge#224](https://ci.openmicroscopy.org/job/FIGURE-merge/224/). See the [console output](https://ci.openmicroscopy.org/job/FIGURE-merge/224/consoleText) for more details.
Possible conflicts:
  - PR #158 dominikl 'Label markdown'
    - static/figure/figure.js
    - static/figure/js/views/right_panel_view.js
  - PR #160 will-moore 'Luts'
    - static/figure/figure.js
    - static/figure/js/views/right_panel_view.js
  - PR #162 dominikl 'Import from json'
    - static/figure/figure.js
  - PR #165 will-moore 'Rois from omero'
    - static/figure/figure.js

**Comment by will-moore** [comment]
Conflicting with #160 LUTs changes to background styles of sliders.
Need to wait for that to be merged to fix conflicts.

**Comment by jburel** [comment]
see also https://github.com/ome/figure/pull/168 for possible conflicts.

**Comment by gusferguson** [comment]
@will-moore 

Tested with http://web-dev-merge.openmicroscopy.org/figure/ user-3 Safari.

Behaves as expected.
Mousewheel scrolling works in Safari.
Good to merge.

File: omero_figure/static/figure/figure.js [hunk 1]
```diff
@@ -1658,6 +1658,153 @@ var UndoView = Backbone.View.extend({
...
+
+    initialize: function(opts) {
+        // This View may apply to a single PanelModel or a list
+        this.models = opts.models;
+        var self = this;
+        this.models.forEach(function(m){
+            self.listenTo(m, 'change:channels', self.render);
+        });
+
+        // this.$el = $("#channel_sliders");
```
Comment (jburel) [inline omero_figure/static/figure/figure.js]: could remove commented out code

File: omero_figure/static/figure/figure.js [hunk 2]
```diff
@@ -1658,6 +1658,153 @@ var UndoView = Backbone.View.extend({
...
+    },
+
+    events: {
+        "keyup .ch_start": "handle_channel_input",
+        // "blur .ch_start": "handle_channel_input",
```
Comment (jburel) [inline omero_figure/static/figure/figure.js]: same

File: omero_figure/static/figure/figure.js [hunk 3]
```diff
@@ -1658,6 +1658,153 @@ var UndoView = Backbone.View.extend({
...
+        "keyup .ch_end": "handle_channel_input",
+        // "blur .ch_end": "handle_channel_input",
```
Comment (jburel) [inline omero_figure/static/figure/figure.js]: same

File: omero_figure/static/figure/figure.js [hunk 4]
```diff
@@ -1658,6 +1658,153 @@ var UndoView = Backbone.View.extend({
...
+            };
+            var reduceFn = function(fn) {
+                return function(prev, curr) {
+                    return fn(prev, curr);
+                }
+            }
+
+            // Comare channels from each Panel Model to see if they are
+            // compatible, and compile a summary json.
+            var chData = this.models.map(function(m){
```
Comment (jburel) [inline omero_figure/static/figure/figure.js]: space before ``{`` not consistent with rest of file

File: omero_figure/static/figure/figure.js [hunk 5]
```diff
@@ -1658,6 +1658,153 @@ var UndoView = Backbone.View.extend({
...
+                return m.get('channels');
+            });
+            // images are compatible if all images have same channel count
+            var allSameCount = chData.reduce(function(prev, channels){
```
Comment (jburel) [inline omero_figure/static/figure/figure.js]: space

**Comment by jburel** [comment]
Thanks for cleanup

**Pull request merged**